File: app/src/main/java/com/example/fragment_part2/MainActivity.java
package com.example.fragment_part2;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity
        implements User1Fragment.MessageSender{
    private User1Fragment user1Fragment;
    private User2Fragment user2Fragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            user1Fragment = new User1Fragment();
            user2Fragment = new User2Fragment();

            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.user1Container, user1Fragment, "USER1")
                    .replace(R.id.user2Container, user2Fragment, "USER2")
                    .commit();
        }
    }

    @Override
    public void sendToUser2(String message) {
        User2Fragment fragment = (User2Fragment) getSupportFragmentManager().findFragmentByTag("USER2");

        if (fragment != null) {
            fragment.receiveMessage(message);
        }
    }

    @Override
    public void sendToUser1(String message) {

        User1Fragment fragment = (User1Fragment) getSupportFragmentManager().findFragmentByTag("USER1");

        if (fragment != null) {
            fragment.receiveMessage(message);
        }
    }
}

File: app/src/main/java/com/example/kotlin_assignment_2/MainActivity.kt
package com.example.kotlin_assignment_2

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val etName = findViewById<EditText>(R.id.etName)
        val etAmount = findViewById<EditText>(R.id.etAmount)
        val spType = findViewById<Spinner>(R.id.spType)
        val etDetails = findViewById<EditText>(R.id.etDetails)
        val btnSubmit = findViewById<Button>(R.id.btnSubmit)

        spType.adapter = ArrayAdapter(
            this,
            android.R.layout.simple_spinner_dropdown_item,
            TransactionType.entries.map { it.name }
        )

        btnSubmit.run {
            text = "Submit Payment"
        }

        val clickHandler by btnSubmit.onDelegateClick()
        clickHandler {
            val name = etName.text.toString()
            val amount = etAmount.text.toString().toDoubleOrNull() ?: 0.0
            val type = TransactionType.valueOf(spType.selectedItem.toString())
            val details = etDetails.text.toString()

            val paymentInfo = PaymentInfo(name, amount, type, details)

            val intent = Intent(this, SummaryActivity::class.java).apply {
                putExtra("PAYMENT_DATA", paymentInfo)
            }
            startActivity(intent)
        }
    }
}